src/daemon/rpmostreed-transaction-types.cxx

```diff
@@ -787,6 +789,50 @@ deploy_has_bool_option (DeployTransaction *self,
   return vardict_lookup_bool (self->options, option, FALSE);
 }
 
+static gboolean
+record_driver_info (const char   *agent,
+                    const char   *sd_unit,
+                    GCancellable *cancellable,
+                    GError      **error)
+{
+  g_auto(GVariantBuilder) caller_info_builder;
+  g_variant_builder_init (&caller_info_builder, G_VARIANT_TYPE ("a{sv}"));
+
+  if (agent)
+    g_variant_builder_add (&caller_info_builder, "{sv}", RPMOSTREE_DRIVER_AGENT_ID,
+                            g_variant_new_string (agent));
+  if (sd_unit)
+    g_variant_builder_add (&caller_info_builder, "{sv}", RPMOSTREE_DRIVER_AGENT_SD_UNIT,
+                           g_variant_new_string (sd_unit));
+
+  g_autoptr(GVariant) composemeta_v = g_variant_builder_end (&caller_info_builder);
+  JsonNode *composemeta_node = json_gvariant_serialize (composemeta_v);
+  glnx_unref_object JsonGenerator *generator = json_generator_new ();
+  json_generator_set_root (generator, composemeta_node);
+
+  char *dnbuf = strdupa (RPMOSTREE_DRIVER_STATE);
+  const char *dn = dirname (dnbuf);
+  if (!glnx_shutil_mkdir_p_at (AT_FDCWD, dn,
```

Comment:
I'd just do `glnx_shutil_mkdir_p_at (AT_FDCWD, RPMOSTREE_RUN_DIR, ...` or so and avoid parsing the target filename.